crates/swc_plugin_runner/src/lib.rs

```diff
@@ -96,14 +97,42 @@ fn load_plugin(plugin_path: &Path, cache: &mut Option<PluginCache>) -> Result<In
 
     return match module {
         Ok(module) => {
-            let import_object = imports! {};
+            let memory = Memory::new(&wasmer_store, MemoryType::new(1, None, false))?;
+            let import_object = imports! {
+                "env" => {
+                    "memory" => memory
+                }
+            };
 
             Instance::new(&module, &import_object).context("Failed to create plugin instance")
         }
         Err(err) => Err(err.into()),
     };
 }
 
+fn copy_memory_to_instance(instance: &Instance, program: &Program) -> Result<(i32, u32), Error> {
+    let alloc = instance.exports.get_native_function::<u32, i32>("alloc")?;
+
+    let serialized = serialize_for_plugin(program)?;
+    let serialized_len = serialized.len();
+
+    let alloc_ptr = alloc.call(serialized_len.try_into()?)?;
+    let memory = instance.exports.get_memory("memory")?;
+    let view = memory.view::<u8>();
+
+    // loop over the Wasm memory view's bytes, assign bytes value of alignedvec from
+    // serialized
+    let ptr_start: usize = alloc_ptr.try_into()?;
+    for (cell, byte) in view[ptr_start..ptr_start + serialized_len + 1]
```

Comment:
I'm not sure about this.
`view` is already a section. Does it require range operation to access?

Reply:
Also, I think this should be something faster like https://doc.rust-lang.org/beta/std/ptr/fn.copy.html

Reply:
I also think there might be some more effecient way, this is mostly copying over from some known examples. Primary focus is `make it working` for now. :sweat_smile: 

I bet most of these code will be refactored anyway while doing further polishings. But having foundational ground something works + having test ensures I can change codes later.